SmellDetector/DTO/MetricsDTO.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SmellDetector.DTO
+{
+    public class MetricsDTO
```

Comment:
Since DTOs are tightly coupled to controllers and the contract they provide to external services, I would package them with the controller, i.e. in the Controllers folder.